action.yml

```
@@ -75,6 +85,17 @@ runs:
          if ${{ inputs.load-all }} = true then
             LoadAllPackages();
          fi;
+
+         OutputLogTo();
+         CloseStream(output_stream);
+
+         # Treat warnings as errors if specified
+         if ${{ inputs.warnings-as-errors }} = 'true' then
```

This can't work, `'true'` is not valid GAP.

I would feel much better if I knew you had actually tested the code, e.g. in your own package repository you can switch out `gap-actions/run-pkg-tests@v2` for `Joseph-Edwards/run-pkg-tests@warnings-as-errors` and then try and see how it behaves.

Then you can also determine whether `${{ inputs.warnings-as-errors }}` is interpolated into the literal `true` (which I suspect is the case) or into a quoted string (and with which quotes), or something else.

---

Note that a few lines earlier we do `if ${{ inputs.load-all }} = true then` so I strongly suspect this needs to be changed to

```suggestion
         if ${{ inputs.warnings-as-errors }} = true then
```